app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/JsonSchemaMigration.java

```diff
@@ -0,0 +1,54 @@
+package com.appsmith.server.migrations;
+
+import com.appsmith.server.domains.ApplicationJson;
+import com.appsmith.server.exceptions.AppsmithError;
+import com.appsmith.server.exceptions.AppsmithException;
+
+public class JsonSchemaMigration {
+    private static Boolean checkCompatibility(ApplicationJson applicationJson) {
```

Comment:
Prefer primitive `boolean` here since we never intend to return `null` here.

```suggestion
    private static boolean checkCompatibility(ApplicationJson applicationJson) {
```